src/BloomExe/Book/Book.cs

```diff
@@ -679,6 +679,96 @@ public void BringBookUpToDate(IProgress progress)
 			}
 		}
 
+		class GuidAndPath
+		{
+			public string Guid; // replacement guid
+			public string Path; // where to find file, relative to distfiles/factoryCollections/Templates
+		}
+
+		private static Dictionary<string, GuidAndPath> _pageMigrations;
+
+		/// <summary>
+		/// Get (after initializing, if necessary) the dictionary mapping page IDs we know how to migrate
+		/// onto the ID and file location of the page we want to update it to.
+		/// Paths are relative to factoryCollections/Templates
+		/// </summary>
+		private static Dictionary<string, GuidAndPath> PageMigrations
+		{
+			get
+			{
+				if (_pageMigrations == null)
+				{
+					_pageMigrations = new Dictionary<string, GuidAndPath>();
+					_pageMigrations["5dcd48df-e9ab-4a07-afd4-6a24d0398382"] = new GuidAndPath() { Guid = "adcd48df-e9ab-4a07-afd4-6a24d0398382", Path = "Basic Book/Basic Book.htm" }; // Basic Text and Picture
+					_pageMigrations["5dcd48df-e9ab-4a07-afd4-6a24d0398383"] = new GuidAndPath() { Guid = "adcd48df-e9ab-4a07-afd4-6a24d0398383", Path = "Basic Book/Basic Book.htm" }; // Picture in Middle
+					_pageMigrations["5dcd48df-e9ab-4a07-afd4-6a24d0398384"] = new GuidAndPath() { Guid = "adcd48df-e9ab-4a07-afd4-6a24d0398384", Path = "Basic Book/Basic Book.htm" }; // Picture on Bottom
+					_pageMigrations["5dcd48df-e9ab-4a07-afd4-6a24d0398385"] = new GuidAndPath() { Guid = "adcd48df-e9ab-4a07-afd4-6a24d0398385", Path = "Basic Book/Basic Book.htm" }; // Just a Picture
+					_pageMigrations["d31c38d8-c1cb-4eb9-951b-d2840f6a8bdb"] = new GuidAndPath() { Guid = "a31c38d8-c1cb-4eb9-951b-d2840f6a8bdb", Path = "Basic Book/Basic Book.htm" }; // Just Text
+					_pageMigrations["FD115DFF-0415-4444-8E76-3D2A18DBBD27"] = new GuidAndPath() { Guid = "aD115DFF-0415-4444-8E76-3D2A18DBBD27", Path = "Basic Book/Basic Book.htm" }; // Picture & Word
+				}
+				return _pageMigrations;
+			}
+		}
+
+		/// <summary>
+		/// Bring the page up to date. Currently this is used to switch various old page types to new versions
+		/// based on Custom Page (so they can actually be customized).
+		/// </summary>
+		/// <param name="page"></param>
+		public void BringPageUpToDate(XmlElement page)
+		{
+			var lineageAttr = page.Attributes["data-pagelineage"];
+			if (lineageAttr == null)
+				return;
+			var lineage = lineageAttr.Value;
+			var originalTemplateGuid = lineage;
+			int index = lineage.IndexOf(";", StringComparison.InvariantCulture);
+			if (index >= 0)
+				originalTemplateGuid = lineage.Substring(0, index);
+			GuidAndPath updateTo;
+			if (!PageMigrations.TryGetValue(originalTemplateGuid, out updateTo))
+				return; // unknown page, don't try to migrate it.
```

Comment:
doesn't this check also cover the case of a page that's already been updated?  the comment seems a bit misleading if I understand the code properly.